waku/v2/node/jsonrpc/filter_api.nim

```diff
@@ -16,10 +17,20 @@ proc installFilterApiHandlers*(node: WakuNode, rpcsrv: RpcServer) =
     messageCache = initTable[ContentTopic, seq[WakuMessage]]()
   
   proc filterHandler(msg: WakuMessage) {.gcsafe, closure.} =
-    debug "WakuMessage received", msg=msg
     # Add message to current cache
-    # @TODO limit max content topics and messages
-    messageCache.mgetOrPut(msg.contentTopic, @[]).add(msg)
+    trace "WakuMessage received", msg=msg
+    
+    # Make a copy of msgs for this topic to modify
+    var msgs = messageCache.getOrDefault(msg.contentTopic, @[])
+
+    if msgs.len >= maxCache:
+      # Message cache on this topic exceeds maximum. Delete oldest.
+      msgs.delete(0,0)
```

Comment:
This seems like it'd be hit quite frequently with a lot of messages coming in. How performant is this if you receive N messages a second and there's a lot of churn? Probably worth profiling first and seeing if it s a hotspot, but might be worth adding a comment here.

Reply:
Will have to profile, but this is probably very inefficient if hit frequently. I will add the comment and consider ways of improving. As I envisioned it, the cache should eventually be dimensioned such that reaching the max size between polling calls would be an infrequent "error" condition. But there is of course no reason to believe clients would always implement it this way.

Reply:
One way to address this would be to have a timed thread that sweeps the map every minute or so, but that would open the "concurrent modification" can of worms. For now I've added comment in da7df23 to profile performance.